### PR TITLE
Add pylintrc #278

### PR DESCRIPTION
**Pull request by hiyashichuka**
I write pylintrc.
The purpose of the pylintrc is to disable vanilla pylint error.

If my pylintrc is against the Google linter, please let me know.

### CODE REVIEW

File: pylintrc [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+# given file (report RP0402 must not be disabled)
+import-graph=
+
+# Create a graph of external dependencies in the given file (report RP0402 must
+# not be disabled)
+ext-import-graph=
+
+# Create a graph of internal dependencies in the given file (report RP0402 must
+# not be disabled)
+int-import-graph=
```
Comment (martindemello) [inline pylintrc]: remove all the entries that are just set to nothing (and any sections that end up being empty)

File: pylintrc [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+dummy-variables-rgx=\*{0,2}(_$|unused_|dummy_)
+
+# List of additional names supposed to be defined in builtins. Remember that
+# you should avoid to define new builtins when possible.
+additional-builtins=
+
+[BASIC]
+
+# List of builtins function names that should not be used, separated by a comma
+bad-functions=map,filter,apply,input,reduce
```
Comment (martindemello) [inline pylintrc]: remove map and filter from this, we use them in pytype.

File: pylintrc [hunk 3]
```diff
@@ -0,0 +1,269 @@
+[MASTER]
```
Comment (martindemello) [inline pylintrc]: this section can probably be removed

**Comment by hiyashichuka** [comment]
> Thanks for your contribution! Could you also add pylint to .travis.yml so that it gets run regularly?

OK
Shall pylint run all .py file?

**Comment by rchen152** [comment]
All .py files except the ones in test_data, I think.

File: pylintrc [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+# not require a docstring
+no-docstring-rgx=(__.*__|main|test.*|.*Test|Module)
+
+# Minimum length for a docstring
+docstring-min-length=10
+
+[MISCELLANEOUS]
+
+# List of note tags to take in consideration, separated by a comma.
+notes=FIXME,XXX,TODO
```
Comment (rchen152) [inline pylintrc]: pytype only uses `TODO` notes, so if you remove `TODO` from this line, then you can also remove `fixme` from the list of disabled messages.

File: pylintrc [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+[FORMAT]
+
+# Maximum number of characters on a single line.
+max-line-length=100
```
Comment (rchen152) [inline pylintrc]: We use a max line length of 80 but allow exceptions for structured pytype comments and urls. I'd suggest:
```
max-line-length=80
ignore-long-lines=(^.{1,80}\s\s# pytype:\s)|(^\s*# http)
```
Note that this catches a real lint error in merge_pyi.py, which I'm planning to push a fix for today, so you'll have to rebase your branch on top of master.

**Comment by hiyashichuka** [comment]
Thank you for pointing it out.
I corrected the problem points.

**Pull request merged**